src/libsyntax/ext/tt/quoted.rs

```diff
@@ -0,0 +1,228 @@
+// Copyright 2017 The Rust Project Developers. See the COPYRIGHT
```

Comment:
We still have `tokenstream::{TokenStream, TokenTree, ...}`.

`quoted::TokenTree` is more general than `tokenstream::TokenTree` -- it also supports "`$`-quoted tokens" from the `macro_rules!` syntax -- `$e`, `$e:expr`, and `$(...)*`. In other words, `quoted::TokenTree` is the parse tree for `macro_rules!` syntax.